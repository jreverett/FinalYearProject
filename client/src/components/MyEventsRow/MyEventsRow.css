.eventrow-event-dropdown {
  padding-left: 0;
  bottom: 0.5rem;
}

.eventrow-event-dropdown .dropdown-toggle:after {
  content: none;
}

#collapsible-nav-dropdown {
  padding: 0.5rem 0.5rem;
}

.eventrow-event-dropdown-delete {
  color: var(--red);
}
