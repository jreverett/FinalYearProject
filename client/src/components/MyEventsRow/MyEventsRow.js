import React, { Component } from 'react';
import { NavDropdown } from 'react-bootstrap';
import { FaEllipsisH } from 'react-icons/fa';
import { eventService } from '../../services';
import { formatDateTime } from '../../utilities';
import '../../common.css';
import './MyEventsRow.css';

class MyEventsRow extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: '',
      start: '',
      removed: false
    };
  }

  componentDidMount() {
    eventService.getEvent(this.props.eventID).then(event => {
      event = event.data;
      this.setState({ title: event.title, start: event.start });
    });
  }

  render() {
    const { title, start, removed } = this.state;
    return (
      <div
        className={`profile-tab-item-container ${removed ? 'hidden' : null}`}
      >
        <div className="profile-tab-item-text-container">
          <p>{`${title} @ ${formatDateTime(start)}`}</p>
        </div>
        <div>
          <NavDropdown
            className="eventrow-event-dropdown"
            title={<FaEllipsisH size={'1.5em'} />}
            id="collapsible-nav-dropdown"
          >
            <NavDropdown.Item href="#action/3.1">
              Send Announcement
            </NavDropdown.Item>
            <NavDropdown.Divider />
            <NavDropdown.Item href="#action/3.4" style={{ color: 'red' }}>
              Delete Event
            </NavDropdown.Item>
          </NavDropdown>
        </div>
      </div>
    );
  }
}

export default MyEventsRow;
