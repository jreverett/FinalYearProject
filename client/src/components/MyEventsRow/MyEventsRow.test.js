import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MyEventsRow from './MyEventsRow';

describe('<MyEventsRow />', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <MyEventsRow />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
