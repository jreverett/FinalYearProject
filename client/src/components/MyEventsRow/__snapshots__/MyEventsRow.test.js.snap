// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyEventsRow /> matches the snapshot 1`] = `
<div
  className="table-item-container null"
  id="eventrow-container"
>
  <div
    className="profile-tab-item-text-container"
  >
    <p>
      loading...
    </p>
  </div>
  <div
    className="eventrow-event-dropdown dropdown nav-item"
    onKeyDown={[Function]}
  >
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle nav-link"
      href="#"
      id="collapsible-nav-dropdown"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <svg
        fill="currentColor"
        height="1.5em"
        size="1.5em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
        />
      </svg>
    </a>
  </div>
</div>
`;
